pkgs/development/interpreters/octave/build-env.nix

```diff
@@ -0,0 +1,83 @@
+{ stdenv, octave, buildEnv
+, makeWrapper, texinfo
+, octavePackages
+, wrapOctave
+, computeRequiredOctavePackages
+, extraLibs ? []
+, extraOutputsToInstall ? []
+, postBuild ? ""
+, ignoreCollisions ? false
+}:
+
+# Create an octave executable that knows about additional packages
+let
+  packages = computeRequiredOctavePackages extraLibs;
+
+in buildEnv {
+    name = "${octave.name}-env";
+    paths = extraLibs ++ [ octave ];
+
+    inherit ignoreCollisions;
+    extraOutputsToInstall = [ "out" ] ++ extraOutputsToInstall;
+
+    buildInputs = [ makeWrapper texinfo wrapOctave ];
+
+    # During "build" we must first unlink the /share symlink to octave's /share
+    # Then, we can re-symlink the all of octave/share, except for /share/octave
+    # in env/share/octave, re-symlink everything from octave/share/octave and then
+    # perform the pkg install.
+    postBuild = ''
+      . "${makeWrapper}/nix-support/setup-hook"
+      # The `makeWrapper` used here is the one defined in
+      # ${makeWrapper}/nix-support/setup-hook
+
+      if [ -L "$out/bin" ]; then
+         unlink $out/bin
+         mkdir -p "$out/bin"
+         cd "${octave}/bin"
+         for prg in *; do
+             if [ -x $prg ]; then
+                makeWrapper "${octave}/bin/$prg" "$out/bin/$prg" --set OCTAVE_SITE_INITFILE "$out/share/octave/site/m/startup/octaverc"
+             fi
+         done
+         cd $out
+      fi
+
+      # Remove symlinks to the input tarballs, they aren't needed.
+      rm $out/*.tar.gz
```

Comment:
That would break the build if a package is a zip.

Reply:
Correct. However, the package that octave compiles with `pkg build` should always produce a tarball. I am not sure about a better way to handle the removal of the compressed compiled package.

Reply:
I did not know that. Then we can leave it like this.